my_variant.h

```diff
@@ -0,0 +1,118 @@
+#include <cassert>
+#include <iostream>
+#include <algorithm>
+#include <type_traits>
+
+using std::cout;
+using std::endl;
+
+
+template <typename... types>
+struct max_size_helper { };
+
+template <typename T, typename... types>
+struct max_size_helper<T, types...> {
+	static size_t constexpr value = std::max(sizeof(T), max_size_helper<types...>::value);
+};
+
+template <> 
+struct max_size_helper<> {
+	static size_t constexpr value = 0;
+};
+
+
+template <typename... types>
+struct max_align_helper { };
+
+template <typename T, typename... types>
+struct max_align_helper<T, types...> {
+	static size_t constexpr value = std::max(alignof(T), max_align_helper<types...>::value);
+};
+
+template <> 
+struct max_align_helper<> {
+	static size_t constexpr value = 0;
+};
+
+/**
+* get index of type from parameter pack
+*/
+
+template <typename T, typename... Ts> struct get_index;
+
+template <typename T, typename... Ts>
+struct get_index<T, T, Ts...> : std::integral_constant<std::size_t, 0> {};
+
+template <typename T, typename Tail, typename... Ts>
+struct get_index<T, Tail, Ts...> :
+    std::integral_constant<std::size_t, 1 + get_index<T, Ts...>::value> {};
+
+template <typename T>
+struct get_index<T> {
```

Comment:
This probably should have been the implementation of the primary template (line 41)? Anyway, static assert here disables any attempt of SFINAE. Is this intentional?

Reply:
So, I am actually not sure what is the right interface. What do we expect for 
get_index<MyClass, double, string, MyOtherClass>::value?

Reply:
Ok, I understand now. Maybe it is better just to remove it and indeed leave option for SFINAE. Will do.